### PR TITLE
Ci pipeline

### PR DESCRIPTION
**Pull request by darahayes**
## Description

ticket: https://issues.jboss.org/browse/AEROGEAR-8020

Adds a basic testing pipeline to the core module. This pipeline clones down both the data-sync-ui and the data-sync-server and uses `npm link` to link up the core module into those repos. This allows us to run the various tests we have for both of those repos.

I also added a linter to this project and included it in the CI.

## Verification Steps

Scroll down and see the green build :)

## Additional Notes

One thing I noticed is that the frontend server has very few tests, and additionally there is some failure happening that is not causing builds to fail. Both here and in the actual frontend project itself. Check out this example build: https://circleci.com/gh/aerogear/data-sync-ui/686

You will see some Sequelize connection error inside the `npm run test` step. I will have to investigate this separately.

Overall, I am concerned that the frontend API server does not really have any integration tests which means user operations on the metadata are not verified.

@wei-lee @StephenCoady @wtrocki I'm not hugely bothered by testing the actual UI itself (right now), but I think we might need to create a ticket to improve the overall quality of tests in the frontend server. What do you think?

### CODE REVIEW

**Comment by wtrocki** [comment]
When moving towards core module and possible future metadata-store module we can unify integration tests in one place and remove code duplication. +100 for creating ticket, but it will be nice to remove code duplication around `user operations on metadata` so we would have one central piece where we could test that etc.

TL;DR - let's create ticket

Edit: Closed by mistake hitting wrong button :)

**Review by wtrocki — APPROVED** [review]

**Comment by wtrocki** [comment]
Me thinking - how server side code coverage works in this way - We do not check node-modules coverage so there is large chunk of code not included in coverage report.
Adding this here may be tricky.

**Comment by darahayes** [comment]
>When moving towards core module and possible future metadata-store module we can unify integration tests

100% agree on this

> but it will be nice to remove code duplication around user operations on metadata

There isn't any duplication around this right now. All operations on the user metadata are done via the data-sync-ui server and I don't think they're being tested properly at all.

> how server side code coverage works in this way

IMO the server coverage should be calculated in the server. The only coverage reporting we do here should be the core module itself.

**Review by StephenCoady — APPROVED** [review]
great PR!
